Sources/Controllers/RouteInfoMenu/OARouteSettingsBaseViewController.mm

```diff
@@ -82,6 +76,29 @@ - (void) didReceiveMemoryWarning
 - (void) viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
+    
+    [self.navigationController setNavigationBarHidden:NO animated:YES];
```

Comment:
The code for styling the navigation bar should be placed in a separate method, configureNavigationBar